arches/app/models/models.py

```diff
@@ -874,15 +894,18 @@ def __init__(self, *args, **kwargs):
     def clean(self):
         if not self.alias:
             Graph.objects.get(pk=self.graph_id).create_node_alias(self)
+        if self.pk == self.source_identifier_id:
```

Comment:
This feels like part of a larger conversation. I would think that we should _never_ get to this point, and that behavior is entirely encapsulated in the `update_from_editable_future_graph` logic. 

And if for some reason it shouldn't be, then this should be added to all entities that have a `source_identifier`

BUT

This may be entirely moot with the editable_future_graph refactor that's in review, so perhaps the conversation should move there 🤔

Reply:
I'm happy to sound this out, but I think it's out of scope, see next comment...